### ISSUE TITLE
Test enhancements

### ISSUE THREAD

**Issue by flea89** [web3-storage/web3.storage#718]
TL;DR 

- Remove smoke mocks of pgrest and cluster in favour of spinning docker images running the services during tests
- Flush DB between tests
- Considering what different layers of testing would look like in Web3.storage ( e2e vs functional vs integration vs unit) and if they're worth investing time in.

---

While writing a test for pinning API endpoints I’ve come across a few opportunities to improve our setup.

At the moment we’re adding tests to the different modules and while they're geared toward been integration kind of tests we're still mocking the db and cluster layer which play down the benefits of testing the overall system.

For instance, in the API package we mock the REST apis calls to the cluster and pgrest that are used behind the hood.

On top of that mocking the responses is difficult and error-prone (we mock the response we want to see, not what will be sent in production).

While I see this approach moves our tests to be slightly closer to integration (rather than unit, with the related benefits) it also inevitably couples them (ie. if we change db or how we interface to it all API tests will fail) without really giving the real benefits of integration testing. It requires the API package to be aware of the DB client implementation and couples it with its implementation, while not really testing the integration with DB and Cluster.

While there might be value in splitting up unit and integration tests, the easiest first step here would be to take the test we already have and, instead of mocking with smoke, to spin out instances of Postgres and IPFS cluster in order to have them be integration-y.

Worth mentioning this is related to miniflare migration given we'll likely have to adapt tests to be ran with miniflare behind the hood.

**Comment by mbommerez** [comment]
This has all been addressed. Closing ticket!
